x/multi-staking/keeper/msg_server.go

```diff
@@ -206,23 +206,28 @@ func (k msgServer) CancelUnbondingDelegation(goCtx context.Context, msg *staking
 		return nil, fmt.Errorf("not allow coin")
 	}
 
-	unbondEntry, found := k.keeper.GetUnbondingEntryAtCreationHeight(ctx, multiStakerAddr, valAcc, msg.CreationHeight)
-	if !found {
-		return nil, fmt.Errorf("unbondEntry not found")
+	unlockID := types.MultiStakingUnlockID(msg.DelegatorAddress, msg.ValidatorAddress)
+	cancelMSCoin, err := k.keeper.DecreaseUnlockEntryAmount(ctx, unlockID, msg.Amount.Amount, msg.CreationHeight)
+	if err != nil {
+		return nil, err
 	}
-	cancelUnbondingCoin := sdk.NewCoin(k.keeper.stakingKeeper.BondDenom(ctx), unbondEntry.Balance)
+
+	cancelBondAmount := cancelMSCoin.BondValue()
```

Comment:
This amount needs to be adjust similarly to `AdjustUnbondAmount`

Reply:
Nah, `AdjustUnbondAmount` should be used when we move shares out from validator, my func is not move any shares, I've just re delegate to the same validator

Reply:
I think we should adjust unbond amount for staking module. We just keep the lock value